src/MacroTools/Extensions/RectangleExtensions.cs

```diff
@@ -21,24 +21,92 @@ public static void AddSound(this Rectangle region, sound soundHandle)
       SetSoundPosition(soundHandle, GetRectCenterX(region.Rect), GetRectCenterY(region.Rect), 0);
       RegisterStackedSound(soundHandle, true, width, height);
     }
-
+    
     /// <summary>
-    /// Renders all units inside the specified <paramref name="area"/> that belong to <paramref name="owningUnitsPlayer"/> invulnerable.
+    /// Creates a group of units inside the specified <paramref name="area"/> that belong to <paramref name="owningUnitsPlayer"/>.
+    /// </summary>
+    /// <param name="area">The area in which to get the group of units.</param>
+    /// <returns>A list of all units found in the specified area</returns>
+    public static List<unit> PrepareUnitsForRescue(this Rectangle area)
```

Comment:
I think what you've with these overloads is great, but I think the API could be cleaner.
Ultimately we only have a few combinations of valid inputs to use. I think instead of having a bunch of parameters, we should just have two: the area, and an Enum that indicates the rescue "mode". The mode would dictate the behaviour. What do you think?

Reply:
I think it is a fantastic way to clean the messiness up